tests/kernel/sched/schedule_api/src/test_sched_timeslice_reset.c

```diff
@@ -143,8 +143,12 @@ void test_slice_reset(void)
 	uint32_t slice_ticks = k_ms_to_ticks_ceil32(SLICE_SIZE);
 	uint32_t half_slice_cyc = k_ticks_to_cyc_ceil32(slice_ticks / 2);
 
-	__ASSERT(slice_ticks % 2 == 0,
-		 "timeslice in ticks much be divisible by two");
+	if (slice_ticks % 2 != 0) {
```

Comment:
Yeah, let's not do this.  The "divisible by two" rule here is a workaround for what amounts to a bug in the test (it's making assumptions about time slice size but doing the math in a foreign unit).  The test logic is only going to work reliably if you follow this weird rule, which is why that assertion exists.

Can we either fix the test to do its math in tick units or change the platform config to not trigger the edge case?

Reply:
Done, half_slice_cyc is added the floating part back that was omitted, when timeslice can't be divisible by two.